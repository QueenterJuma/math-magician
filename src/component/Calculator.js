import React, { useState } from 'react';
import './style/Calculator.css';
import calculate from '../Logic/calculate';

const Calculator = () => {
  const [total, setTotal] = useState(0);
  const [operation, setOperation] = useState(null);
  const [next, setNext] = useState(null);
  const handleClick = (e) => {
    const result = calculate({ total, operation, next }, e.target.textContent);
    setTotal(result.total);
    setOperation(result.operation);
    setNext(result.next);
  };

  return (
    <div id="calculator">
      <div className="display">
        {total}
        {operation}
        {next}
      </div>
      <div id="keypad">
        <button onClick={handleClick} type="button">
          AC
        </button>
        <button onClick={handleClick} type="button">
          +/-
        </button>
        <button onClick={handleClick} type="button">
          %
        </button>
        <button onClick={handleClick} type="button" className="orange">
          รท
        </button>
        <button onClick={handleClick} type="button">
          7
        </button>
        <button onClick={handleClick} type="button">
          8
        </button>
        <button onClick={handleClick} type="button">
          9
        </button>
        <button onClick={handleClick} type="button" className="orange">
          x
        </button>
        <button onClick={handleClick} type="button">
          4
        </button>
        <button onClick={handleClick} type="button">
          5
        </button>
        <button onClick={handleClick} type="button">
          6
        </button>
        <button onClick={handleClick} type="button" className="orange">
          -
        </button>
        <button onClick={handleClick} type="button">
          1
        </button>
        <button onClick={handleClick} type="button">
          2
        </button>
        <button onClick={handleClick} type="button">
          3
        </button>
        <button onClick={handleClick} type="button" className="orange">
          +
        </button>
        <button onClick={handleClick} type="button" className="zero">
          0
        </button>
        <button onClick={handleClick} type="button">
          .
        </button>
        <button onClick={handleClick} type="button" className="orange">
          =
        </button>
      </div>
    </div>
  );
};

export default Calculator;
