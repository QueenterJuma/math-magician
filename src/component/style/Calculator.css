#calculator {
  width: 400px;
  height: 420px;
  background-color: #858693;
  margin: 2rem auto;
  box-shadow: 0 0 10px rgba(99, 77, 77, 0.5);
}

.display {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
  height: 15%;
  color: #e0e0e0;
  font-size: 1.5rem;
  text-align: end;
}

.display span {
  font-size: 50px;
}

#keypad {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(5, 1fr);
  width: 100%;
  height: 87%;
}

.keypad button {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  border: none;
  background-color: #eee;
  font-size: 3rem;
  font-weight: bold;
  color: #222020;
  cursor: pointer;
  transition: background-color 0.2s;
}

.orange {
  background-color: #f5913f;
}

.keypad button:hover {
  background-color: #979090;
}

.zero {
  grid-column: 1 / span 2;
}
