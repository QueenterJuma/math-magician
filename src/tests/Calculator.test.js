import React from 'react';
import renderer from 'react-test-renderer';
// eslint-disable-next-line
import Calculator from '../Component/Calculator';

describe('Calculator component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
