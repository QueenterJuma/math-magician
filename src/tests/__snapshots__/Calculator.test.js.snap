// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly 1`] = `
<div
  className="math"
>
  <h2
    className="cal"
  >
    Let's do some Maths
  </h2>
  <div
    id="calculator"
  >
    <div
      className="display"
    >
      0
    </div>
    <div
      id="keypad"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
